flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -351,38 +350,67 @@ ByteBuf write(ByteBufAllocator allocator) throws IOException {
 
 				CompositeByteBuf composityBuf = allocator.compositeDirectBuffer();
 				composityBuf.addComponent(headerBuf);
-				composityBuf.addComponent(buffer);
+				composityBuf.addComponent(buffer.asByteBuf());
 				// update writer index since we have data written to the components:
-				composityBuf.writerIndex(headerBuf.writerIndex() + buffer.writerIndex());
+				composityBuf.writerIndex(headerBuf.writerIndex() + buffer.asByteBuf().writerIndex());
 				return composityBuf;
 			}
 			catch (Throwable t) {
 				if (headerBuf != null) {
 					headerBuf.release();
 				}
-				buffer.release();
+				buffer.recycleBuffer();
 
 				ExceptionUtils.rethrowIOException(t);
 				return null; // silence the compiler
 			}
 		}
 
-		static BufferResponse readFrom(ByteBuf buffer) {
-			InputChannelID receiverId = InputChannelID.fromByteBuf(buffer);
-			int sequenceNumber = buffer.readInt();
-			int backlog = buffer.readInt();
-			boolean isBuffer = buffer.readBoolean();
-			boolean isCompressed = buffer.readBoolean();
-			int size = buffer.readInt();
+		/**
+		 * Parses the message header part and composes a new BufferResponse with an empty data buffer. The
+		 * data buffer will be filled in later. This method is used in credit-based network stack.
```

Comment:
We do not need to emphasis `This method is used in credit-based network stack.` which can be removed.

Reply:
Removed this sentence.